docs/usage/compute/machine.md

```diff
@@ -1 +1,69 @@
+- machinePoolRef (`string`): machinePoolRef defines the machine pool to run the machine in. If empty, a scheduler will figure out an appropriate pool to run the machine in.
+- image (`string`): image is the optional URL providing the operating system image of the machine.
+- volumes (`list`): volumes are list volumes(storage) attached to this machine.
+- networkInterfaces (`list`): networkInterfaces define a list of network interfaces present on the machine
+- ignitionRef (`string`): ignitionRef is a reference to a `secret` containing the ignition YAML for the machine to boot up. If a key is empty, `DefaultIgnitionKey` will be used as a fallback. (`Note`: Refer to <a href="https://github.com/ironcore-dev/ironcore/tree/main/config/samples/e2e/bases/ignition">Sample Ignition</a> for creating ignition secret)
+
+
+## Reconciliation Process
+
+1. **Machine Scheduling**: 
+The `MachineScheduler` controller continuously watches for `Machines` without an assigned `MachinePool` and tries to schedule it on available and matching MachinePool.
+  - **Monitor Unassigned Machines**: The scheduler continuously watches for machines without an assigned `machinePoolRef`.
+  - **Retrieve Available Machine Pools**: The scheduler fetches the list of available machine pools from the cache.
+  - **Make Scheduling Decisions**: The scheduler selects the most suitable machine pool based on resource availability and other policies.
+  - **Update Cache**: The scheduler updates the cache with recalculated allocatable `machineClass` quantities.
+  - **Assign MachinePoolRef**: The scheduler assigns the selected `machinePoolRef` to the machine object.
+
+2. **IRI Machine creation**: Once the Machine is allocated to a particular pool, the `MachineController` processes the `Machine` resource and it extracts the `ignitionData`, `networkInterfaces` and `volumes` information from the `spec` and prepares IRI machine resource.
+
+3. **Machine Brokering**: Once IRIMachine object is prepared create/update the machine request is sent to a broker via the IRI interface(via GRPC call). An actual VM is created when the request reaches the compute provider. Once the response is received from IRI call Machine status is updated with the status recieved.
+
+4. **Network Interface handling**: `MachineControllerNetworkinterface` takes care of attaching/detaching Network interfaces defined for the machine. Once the attachment is successful status is updated from `Pending` to `Attached`.
+
+5. **Volume handling**: `MachineControllerVolume` takes care of attach/detach of Volumes(Storage) defined for machine. Once the attachment is successful status is updated from `Pending` to `Attached`.
+
+6. **Ephemeral resource handling**: If `NetworkIntreface` or `Volume` is defined as ephemeral(i.e. coupled to the lifetime of the machine object) in the machine spec, `MachineEphemeralControllers` takes care of creating and destroying respective objects on creation/deletion of the machine. 
+
```

Comment:
Maybe it also makes sense here to mention that you can bind the `Volume` and NIC lifecycle with the `Machine` lifecycle by creating ephemeral resources on the fly and reference here the e2e examples for those.